### PR TITLE
`Iris`: Call API to accept Iris policy

### PR DESCRIPTION
**Pull request by dakennguyen**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [ ] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [ ] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
Call API to accept Iris policy and get the accepted timestamp from the server

### Steps for Testing

Prerequisites:
- 1 Student that hasn't accepted Iris policy yet
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Go to a programming exercise with Iris enabled
3. Click on Iris icon and should be prompted to accept the policy
4. Accept the policy and a welcome message will be shown
5. Reload the page and the welcome message should still be there
6. Chat something
7. Reload the page and the welcome message should be disappeared

#### Performance Review
- [ ] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by SolizerCodes — DISMISSED** [review]
I tested on our test server and I couldn't get the accept-dialog again after accepting once. Looks good 👍

**Review by RY997 — CHANGES_REQUESTED** [review]
After user accept the policy and open a new exercise, there is no first message. I need the `userAccepted` in `onInit()` can be read as true, so it's better to make it async or add a timer.
image

File: src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts
Comment (dakennguyen) [inline src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:142]: @RY997 by calling to load first message here I think it will solve the problem too

Comment (RY997) [reply src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:142]: The messages should be the value in stateStore which is after this part, so I'm not sure if this works.

Comment (dakennguyen) [reply src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:142]: @RY997 ah then I'll leave it to your PR then. I think we can merge it like this now if there are no bugs

Comment (RY997) [reply src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:142]: I plan to test once and if there's no bugs, we can merge

**Review by SabrinaGlatz — CHANGES_REQUESTED** [review]
Tested on Ts9. It works as expected, but I can see still the accept policy message for less than a second when I open Iris in another programming exercise (see video). It would be good if it won't be visible at all.

https://github.com/ls1intum/Artemis/assets/125522114/d57f192d-0469-4882-a3a4-48c3db952769

**Review by RY997 — APPROVED** [review]
Code LGTM and also tested locally, worked well.

File: src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts
Comment (RY997) [inline src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:164]: I think this is for autofocus, maybe you mis-deleted this

Comment (dakennguyen) [reply src/main/webapp/app/iris/exercise-chatbot/exercise-chatwidget/exercise-chat-widget.component.ts:164]: right!! thank you, I reverted it

File: src/main/webapp/app/core/user/user.service.ts
Comment (JohannesStoehr) [inline src/main/webapp/app/core/user/user.service.ts:73]: This should return an observable of dayjs like getLatestDueDate() does

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by eceeeren — DISMISSED** [review]
Tested on TS9, works well!

**Review by asliayk — APPROVED** [review]
tested on ts9, works fine!

**Pull request merged**
